graylog2-web-interface/src/components/alertconditions/AlertConditionTestModal.jsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Alert, Button, Modal } from 'react-bootstrap';
+
+import { BootstrapModalWrapper } from 'components/bootstrap';
+import { Spinner } from 'components/common';
+import CombinedProvider from 'injection/CombinedProvider';
+import style from './AlertConditionTestModal.css';
+
+const { AlertConditionsActions } = CombinedProvider.get('AlertConditions');
+
+class AlertConditionTestModal extends React.Component {
+  static propTypes = {
+    stream: PropTypes.object.isRequired,
+    condition: PropTypes.object.isRequired,
+  };
+
+  state = {
+    testResults: undefined,
+    isTesting: false,
+  };
+
+  open = () => {
+    this.modal.open();
+    this.testCondition();
+  };
+
+  close = () => {
+    this.modal.close();
+  };
+
+  testCondition = () => {
+    this.setState({ isTesting: true, testResults: undefined });
+    AlertConditionsActions.test(this.props.stream.id, this.props.condition.id)
+      .then(
+        testResults => this.setState({ testResults: testResults }),
+        (error) => {
+          if (error.status === 400) {
+            // Condition testing failed but we should still get results in the body
+            this.setState({ testResults: error.additional.body });
+            return;
+          }
+          // Create a default error message to display in frontend
+          this.setState({
+            testResults: {
+              error: true,
+              error_messages: [{
+                type: 'Unexpected error',
+                message: 'Could not test Condition, please try again or check your server logs for more information.',
+              }],
+            },
+          });
+        },
+      )
+      .finally(() => this.setState({ isTesting: false }));
+  };
+
+  renderErroneousCondition = (testResults) => {
+    return (
+      <span>
+        <p><b>There was an error testing the Condition.</b></p>
+        <p>
+          <ul className={style.errorMessages}>
+            {testResults.error_messages.map(({ message, type }) => (
+              <li key={`${type}-${message}`}>{message} ({type})</li>
+            ))}
+          </ul>
+        </p>
+      </span>
+    );
+  };
+
+  renderSatisfiedCondition = (testResults) => {
+    return (
+      <span>
+        <p>Condition was satisfied and an Alert would be triggered.</p>
+        <p><b>Details</b>: {testResults.description}</p>
+      </span>
```

Comment:
The background color for the "condition satisfied" state is the same as for the "condition not satisfied". This makes it hard to differentiate between the two. I had to look twice to see that my condition matched after changing the query because I somehow expected a different visual. :wink: 

![image](https://user-images.githubusercontent.com/461/49164776-16b8f500-f330-11e8-80cc-b3ca922e8765.png)

![image](https://user-images.githubusercontent.com/461/49164808-26d0d480-f330-11e8-80d5-bb628dabfa8f.png)

Do you think it would make sense to use different background colors for the two states?

Reply:
We could use different colours, but usually each colour has a connotation: green is "good", yellow is "meh", red is "bad".
On this case we already use red when the condition throws an error. These two cases are a bit more complicated because none of them really fit into any of the cases above, it's more like testing an extractor (where we also use blue afaik).
I can play a bit with it and see if we can do something about it.

Reply:
Okay, makes sense. Let's keep it as it is for now. Thank you!

Reply:
I added a small icon next to it, to have at least a visual hint.

Reply:
Thank you